src/ldp/http/SimpleBodyParser.ts

```diff
@@ -0,0 +1,52 @@
+import { BodyParser } from './BodyParser';
+import { HttpRequest } from '../../server/HttpRequest';
+import { Quad } from 'rdf-js';
+import { QuadRepresentation } from '../representation/QuadRepresentation';
+import { RepresentationMetadata } from '../representation/RepresentationMetadata';
+import { StreamParser } from 'n3';
+import { TypedReadable } from '../../util/TypedReadable';
+import { UnsupportedMediaTypeHttpError } from '../../util/errors/UnsupportedMediaTypeHttpError';
+import 'jest-rdf';
+
+export class SimpleBodyParser extends BodyParser {
+  private static readonly contentTypes = [
+    'application/n-quads',
+    'application/trig',
+    'application/n-triples',
+    'text/turtle',
+    'text/n3',
+  ];
+
+  public async canHandle(input: HttpRequest): Promise<void> {
+    const contentType = input.headers['content-type'];
+
+    if (contentType && !SimpleBodyParser.contentTypes.some((type): boolean => contentType.includes(type))) {
+      throw new UnsupportedMediaTypeHttpError('This parser only supports RDF data.');
+    }
+  }
+
+  public async handle(input: HttpRequest): Promise<QuadRepresentation> {
+    const contentType = input.headers['content-type'];
+
+    if (!contentType) {
+      return undefined;
+    }
+
+    const specificType = contentType.split(';')[0];
+
+    const metadata: RepresentationMetadata = {
+      raw: [],
+      profiles: [],
+      contentType: specificType,
+    };
+
+    // StreamParser is a Readable but typings are incorrect at time of writing
```

Comment:
Maybe we should invest some time in fixing those RDFJS typings?
I'm sure this will haunt us for a long time otherwise...

Reply:
Can also look into definitely typed for this then (sort of didn't want to, but I agree it's probably best). This would require changes in both rdf-js and n3 typings right?

Reply:
> This would require changes in both rdf-js and n3 typings right?

I think so, yes. Probably also some of my TS parsers then, but I'll look into that when we get there :-)